src/packages/cli/src/detach.ts

```diff
@@ -0,0 +1,259 @@
+ * @returns boolean indicating whether the instance was found.
+ */
+export async function stopDetachedInstance(
+  instanceName: string
+): Promise<boolean> {
+  const instance = await findDetachedInstanceByName(instanceName);
+  if (instance !== undefined) {
+    try {
+      process.kill(instance.pid, "SIGTERM");
+    } catch (err) {
+      // process.kill throws if the process was not found (or was a group process in Windows)
+      return false;
+    } finally {
+      removeDetachedInstanceFile(instance.pid);
+    }
+    return true;
+  }
+  return false;
+}
+
+/**
+ * Start an instance of Ganache in detached mode.
+ * @param  {string[]} argv arguments to be passed to the new instance.
+ * @returns {Promise<DetachedInstance>} resolves to the DetachedInstance once it
+ * is started and ready to receive requests.
+ */
+export async function startDetachedInstance(
+  module: string,
+  args: StartArgs<FlavorName>,
+  version: string
+): Promise<DetachedInstance> {
+  const flavor = args.flavor;
+  const childArgs = createFlatChildArgs(args);
+  const child = fork(module, childArgs, {
+    stdio: ["ignore", "ignore", "pipe", "ipc"],
+    detached: true
+  });
+  const cmd = (await psList()).find(p => p.pid === child.pid).cmd;
+
+  // Any messages output to stderr by the child process (before the `ready`
+  // event is emitted) will be streamed to stderr on the parent.
+  child.stderr.pipe(process.stderr);
+
+  const instances = await getDetachedInstances();
+  const instanceNames = instances.map(instance => instance.instanceName);
+
+  let instanceName: string;
+  do {
+    instanceName = createInstanceName();
+  } while (instanceNames.indexOf(instanceName) !== -1);
+
+  await new Promise<void>((resolve, reject) => {
+    child.on("message", message => {
+      if (message === READY_MESSAGE) {
+        resolve();
+      }
+    });
+
+    child.on("error", err => {
+      // This only happens if there's an error starting the child process, not
+      // if Ganache throws within the child process.
+      console.error(`${START_ERROR}\n${err.message}`);
+      process.exitCode = 1;
+      reject(err);
+    });
+
+    child.on("exit", (code: number) => {
+      // This shouldn't happen, so ensure that we surface a non-zero exit code.
+      process.exitCode = code === 0 ? 1 : code;
+      reject(
+        new Error(
+          `${START_ERROR}\nThe detached instance exited with error code: ${code}`
+        )
+      );
+    });
+  });
+
+  // destroy the ReadableStream exposed by the child process, to allow the
+  // parent to exit gracefully.
+  child.stderr.destroy();
+  child.unref();
+  child.disconnect();
+
+  const instance: DetachedInstance = {
+    startTime: Date.now(),
+    pid: child.pid,
+    instanceName,
+    host: args.server.host,
+    port: args.server.port,
+    flavor,
+    cmd,
+    version
+  };
+
+  const instanceFilename = `${dataPath}/${instance.pid}`;
+
+  writeFileSync(instanceFilename, JSON.stringify(instance));
```

Comment:
I believe this will throw in the event that the filesystem/file is readonly or otherwise restricted in access.

Reply:
Definitely not ideal! I've tested a couple of scenarios here: #3904, and although some of the cases are sub-optimal, I don't think any are critical.

Most importantly:
- If a running instance's file cannot be _read_, the instance is terminated
- if a running instance's file cannot be _deleted_, it is just ignored

Reply:
Cool Cool, 

Yeah the only reason it is in my head is because of my recent NTFS issues flipping between OS's. I keep my `/home` on a separate partition. Sometimes symlinking can be involved and things just aren't available. 

Once the FS is involved I get a bit hand wavey using `try catch`